### PR TITLE
[#1510] Commodities list: Lent out toolbar filter (BE + FE)

### LINKED ISSUE #1510: Toolbar filter "Lent out" on /commodities list page (follow-up #1452)
## Motivation

#1452 base AC line:

> List page badges currently-lent items; toolbar filter exposes "lent out".

The badge half landed in #1506. The toolbar filter half was deferred — `/lent` exists as a dedicated page and absorbs most of the "find what's lent" intent, so this is convenience-only. Worth filing so the original AC isn't silently dropped.

### Backend

- Extend the existing `GET /api/v1/g/{slug}/commodities` filter set with a `lent_out=true|false` query param (or fold into the existing status filter as a synthetic value, mirroring the derived-status decision from #1452).
- Implementation: subquery or LEFT JOIN against `commodity_loans` filtered to `returned_at IS NULL`. Keep it on the existing pagination / sort path so axe + keyset behavior doesn't drift.

### Frontend

- Add a "Lent out" checkbox / chip to the toolbar filter row on `frontend/src/pages/commodities/CommoditiesListPage.tsx`. Same component shape as the existing filter chips.
- When active, badge already renders on each row from #1506; no double-rendering required.
- i18n key in `frontend/src/i18n/locales/{en,cs,ru}/commodities.json` matching the existing toolbar pattern.

### Tests

- BE: registry-level test for the new filter, including edge cases (commodity with multiple loans where the latest is returned but an older one is open — only the open-row signal counts).
- FE: vitest covers the filter wiring; jest-axe clean.
- E2E: lend an item → toggle filter → only lent items visible → return → toggled-on filter empties.

## Out of scope

- Multi-state combined filter ("lent OR sold"). Single boolean is enough — users either care about "currently lent" or they don't.
- Stacking with a future "in service" filter (#1508). When that lands, both can coexist as separate chips.

## Acceptance criteria

- [ ] BE filter param shipped + tested.
- [ ] FE chip in the toolbar; active state, axe clean.
- [ ] E2E covers the round-trip.
- [ ] i18n in en + cs + ru.

## Cross-refs

- Parent feature: #1452 / PR #1506.
- Sibling: #1508 (in-service filter chip will follow the same shape).
- Parent epic: #1397.

### PR DESCRIPTION
**Pull request by denisvmedia**
Closes #1510 — the toolbar half of the original #1452 acceptance line (badge half landed in #1506).

## Summary

- **BE**: `GET /api/v1/g/{slug}/commodities` accepts `lent_out=true|false`. The apiserver pre-resolves the open-loan commodity IDs from `CommodityLoanRegistry` and passes them through `CommodityListOptions` so the memory backend stays decoupled from loans; postgres uses an `EXISTS` subquery against `commodity_loans` (partial index `idx_commodity_loans_active` keeps it cheap). Closed (returned) loans never flag a commodity as lent — predicate is `returned_at IS NULL`.
- **FE**: Boolean chip after the warranty filter on `/commodities`, mirroring the existing inactive toggle's button shape. URL state via `lent_out=1`; `useCommodities` + query-key carry `lentOut` so TanStack Query refetches on toggle. Clear-filters + `hasFilters` both wire it through.
- **i18n**: `en` key added (`commodities:filter.lentOut: "Lent out"`); cs/ru remain `{}` stubs per project convention.

## Acceptance criteria

- [x] BE filter param shipped + tested (memory registry table + apiserver round-trip).
- [x] FE chip in the toolbar, active state with `aria-pressed`.
- [x] E2E covers the round-trip (lend → toggle → only lent visible → return → empty).
- [x] i18n in en (cs/ru intentional `{}` per repo convention).

## Test plan

- [x] `go test ./registry/... ./apiserver/...` — green
- [x] `npm run typecheck` (frontend) — green
- [x] `npm run lint` (frontend) — green
- [x] `npx prettier --check` on touched files — clean
- [x] `npx vitest run` — 119 files / 829 tests pass (1 new case)
- [x] `golangci-lint run` on changed packages — 0 issues
- [ ] Playwright e2e `commodity-lent-out-filter.spec.ts` — exercised on CI (host docker stack not running locally)

## Out of scope

- Multi-state combined filter ("lent OR sold"). Single boolean is enough.
- `in_service` chip (#1508) will follow the same shape.

### CODE REVIEW

File: e2e/tests/commodity-lent-out-filter.spec.ts [hunk 1]
```diff
@@ -0,0 +1,154 @@
...
+      await expect(chip).toHaveAttribute('aria-pressed', 'true')
+      await expect(idleCard).toHaveCount(0, { timeout: 15000 })
+      await expect(lentCard).toBeVisible()
+
+      // Mark the loan returned via the API. The list query is
+      // invalidated by useReturnLoan in production, but we're poking
+      // the BE directly here — the chip stays toggled, and a manual
+      // refresh of the list URL re-fetches with the same filter.
+      const returnResp = await request.patch(
+        `/api/v1/g/${encodeURIComponent(group.slug)}/commodities/${encodeURIComponent(lent.id)}/loans/${encodeURIComponent(loanID!)}`,
```
[the remark this answers is not in the document]
Comment (denisvmedia) [reply e2e/tests/commodity-lent-out-filter.spec.ts]: Good catch — confirmed PATCH only mutates borrower fields + `due_back_at`; `returned_at` there is silently dropped, so the post-return assertions were effectively re-asserting the pre-return state. Fixed in 03a9166 — the test now closes the loan via `POST .../loans/{loanID}/return` (empty body → BE defaults `returned_at` to today).

File: go/apiserver/commodities.go [hunk 2]
```diff
@@ -156,9 +172,41 @@ func parseCommodityListOptions(q url.Values) registry.CommodityListOptions {
...
+		// is true; everything else is false). Presence alone activates the
+		// filter — empty/missing leaves opts.LentOut nil (no filter).
+		b := v == "1" || strings.EqualFold(v, "true")
+		opts.LentOut = &b
+	}
 	return opts
 }
 
+// listOpenLoanCommodityIDs collects every commodity ID in the current
+// group whose latest loan is still open (returned_at IS NULL). The
```
[the remark this answers is not in the document]
Comment (denisvmedia) [reply go/apiserver/commodities.go]: Agreed — the comment was sloppy. Fixed in 03a9166: it now says "every commodity ID in the current group that has at least one open loan (a `commodity_loans` row with `returned_at IS NULL`)", which matches both the dedupe-by-`commodity_id` behaviour and the predicate the postgres `EXISTS` subquery enforces.

File: go/apiserver/commodities.go [hunk 3]
```diff
@@ -156,9 +172,41 @@ func parseCommodityListOptions(q url.Values) registry.CommodityListOptions {
...
+	for _, l := range loans {
+		if l == nil {
+			continue
+		}
+		if _, dup := seen[l.CommodityID]; dup {
+			continue
+		}
+		seen[l.CommodityID] = struct{}{}
+		ids = append(ids, l.CommodityID)
+	}
```
[the remark this answers is not in the document]
Comment (denisvmedia) [reply go/apiserver/commodities.go]: Fixed in 03a9166. Switched to a paginating loop (1000 rows per page; stop when `offset >= total` or the page comes back empty as a defensive guard against a buggy backend reporting `total > rows-available`). Personal-inventory groups will exit on the first page in practice, but the filter is now correct for arbitrarily large groups too.

File: e2e/tests/commodity-lent-out-filter.spec.ts [hunk 4]
```diff
@@ -0,0 +1,154 @@
...
+import {
+  createCommodityViaAPI,
+  deleteCommodityViaAPI,
+  ensureLocationAndArea,
+  extractApiAuth,
+  resolveActiveGroup,
+} from './includes/commodities-api.js'
+
+test.describe('Commodities list — Lent out filter (#1510)', () => {
+  test('toggle filter shows only currently-lent items, returns flush them', async ({
```
[the remark this answers is not in the document]
Comment (denisvmedia) [reply e2e/tests/commodity-lent-out-filter.spec.ts]: Fixed in 03a9166.

File: go/registry/postgres/commodities.go [hunk 5]
```diff
@@ -384,6 +390,26 @@ func buildCommodityWhere(opts registry.CommodityListOptions) (string, []any) {
...
+		return ""
+	}
+	op := "EXISTS"
+	if !*lentOut {
+		op = "NOT EXISTS"
+	}
+	return fmt.Sprintf(
+		"%s (SELECT 1 FROM %s WHERE commodity_id = commodities.id AND returned_at IS NULL)",
+		op, loansTable,
+	)
```
[the remark this answers is not in the document]
Comment (denisvmedia) [reply go/registry/postgres/commodities.go]: Good catch. Fixed in 938d0f5 — `buildCommodityWhere` now takes `commoditiesTable` alongside `loansTable`, and `buildLentOutCond` uses `commoditiesTable + ".id"` in the correlated subquery. `ListPaginated` threads `r.tableNames.Commodities()` next to `r.tableNames.CommodityLoans()` through the call chain, so any `TableNames` override (schema prefix, sharded suffix, test stub) now flows end-to-end into the subquery's outer reference.

File: go/apiserver/commodities.go [hunk 6]
```diff
@@ -61,6 +62,21 @@ func (api *commoditiesAPI) listCommodities(w http.ResponseWriter, r *http.Reques
...
+	// cheap — partial index `idx_commodity_loans_active` keeps the read
+	// to the tiny working set of currently-lent rows.
+	if opts.LentOut != nil && regSet.CommodityLoanRegistry != nil {
+		ids, err := listOpenLoanCommodityIDs(r.Context(), regSet.CommodityLoanRegistry)
+		if err != nil {
+			internalServerError(w, r, err)
+			return
+		}
+		opts.OpenLoanCommodityIDs = ids
+	}
```
[the remark this answers is not in the document]
Comment (denisvmedia) [reply go/apiserver/commodities.go]: Agreed — on postgres that pre-fetch was pure overhead. Fixed in 938d0f5 via a capability marker: new `registry.NativeLentOutFilterer` interface with a no-op `SupportsNativeLentOutFilter()` method, implemented by the postgres `CommodityRegistry` (whose `EXISTS` subquery already does the join in one query). The apiserver type-asserts and skips the pre-fetch:

```go
if _, native := commodityReg.(registry.NativeLentOutFilterer); !native {
    // pre-resolve OpenLoanCommodityIDs from CommodityLoanRegistry
}
```

The memory backend doesn't implement the interface, so the pre-resolve path stays for it. No change to the public `CommodityRegistry` signature — the capability is strictly opt-in via the assertion.

File: go/registry/memory/commodities.go [hunk 7]
```diff
@@ -330,9 +330,24 @@ func commodityMatches(c *models.Commodity, opts registry.CommodityListOptions, q
...
+// open-loan ID set. Returns true when LentOut is nil (no filter) so the
+// caller can compose this with the other branches without an additional
+// nil-guard. Split out of commodityMatches to keep that function under
+// the gocyclo threshold.
+func matchesLentOut(c *models.Commodity, lentOut *bool, openIDs []string) bool {
+	if lentOut == nil {
+		return true
+	}
+	return slices.Contains(openIDs, c.ID) == *lentOut
+}
```
[the remark this answers is not in the document]
Comment (denisvmedia) [reply go/registry/memory/commodities.go]: Fair point. Fixed in 938d0f5 — `filterCommodities` lifts `opts.OpenLoanCommodityIDs` into a `map[string]struct{}` once per call and passes it through `commodityMatches` to `matchesLentOut`. Per-row check is now O(1) instead of O(#openIDs); overall filter is O(#commodities) instead of O(#commodities × #openIDs). The previous `slices.Contains` form would have started biting as the open-loan set grew; it scales now.
